backend/mem/file.go

```diff
@@ -0,0 +1,469 @@
+package mem
+
+import (
+	"errors"
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/utils"
+	"io"
+	"path"
+	"time"
+)
+
+//File implements vfs.File interface for os fs.
+type File struct {
+	exists    bool
+	timeStamp time.Time
+	isOpen    bool
+	isRef     bool         //has it been referenced before?
+	privSlice []byte       //the file contents
```

Comment:
Rather than naming this after the data structure ie: what it means to the language (it is literally a private slice) name this (and other variables) for what it means to us (the developers that will be writing and maintaining the code). According to the comment maybe a more appropriate name for this variable is 'contents'? 'File.contents' is a lot more meaningful than 'File.privSlice'. 

Similarly above, if someone sees 'file.isRef' somewhere, will they wonder whether this means a boolean to refer to whether the file *is* a reference vs whether it has been referenced? Does 'isReferenced' make more sense? I'm literally asking, not trying to lead you to my line of thinking, I don't actually know yet how 'isRef' is used.

And 'timeStamp'... yep, it is a time.Time. What does this timeStamp mean to the context of the file from our point of view when we see this property referenced elsewhere in the code? Is it a created timestamp? modified/last accessed timestamp? Any touch to the file (all of the above?) in which case would something like file.lastTouched, or file.lastModified be more descriptive?

I know I asked you to comment these properties in my last review, I should have thought about what I was requesting a little more. If you have to comment property/member names to describe what they are, it is likely a sign that the name needs to be rethought. The comments will help when someone is looking at this struct definition, but a good name will be much more useful throughout the code.